.github/workflows/publish.yml

```
@@ -94,8 +94,18 @@ jobs:
 
   snapshot:
-    if: ${{ github.event_name == 'workflow_dispatch' }}
-    name: Publish snapshot/beta to npm
+    # Guard against publishing snapshots from the protected release branch.
+    # Both github.ref (the branch selected in the UI) and inputs.branch (the
+    # free-text checkout ref) must be checked, since they are independent values
+    # and the checkout step uses inputs.branch directly.
+    if: >-
+      ${{
+        github.event_name == 'workflow_dispatch' &&
+        github.ref != 'refs/heads/changeset-release/main' &&
```

actionlint (1.7.12)

[error] property "branch" is not defined in object type {npm_access: string; npm_tag: string; snapshot_tag: string}

(expression)